src/api/notifications/notifications.service.ts

```diff
@@ -0,0 +1,73 @@
+import { checkInDBSchema } from "../../models/event";
+import { getDbClient } from "../../utils/database";
+import { getDiscordBot } from "../../utils/discord";
+import { notificationsRequest } from "./notifications.schema";
+import { Response } from "express";
+import { Client, MessageEmbed } from "discord.js";
+import { CONSTANTS } from "../../utils/constants";
+
+export interface getDiscordIDSchema {
+  userIDArray: Array<string>;
+  failedEmails: Array<string>;
+  successEmails: Array<string>;
+}
+
+export const getDiscordID = async (emailArray: Array<string>) => {
+  const db = (await getDbClient()).db().collection(`jack-notifications`);
+  let userIDArray: string[] = [];
+  let failedEmails: string[] = [];
+  let successEmails: string[] = [];
+  try {
+    for (let index = 0; index < emailArray.length; index++) {
+      const user = await db.findOne<checkInDBSchema>({
+        email: emailArray[index],
+      });
+      if (!user) {
+        failedEmails.push(emailArray[index]);
+      } else {
+        const userId = user.discordID;
+        userIDArray.push(userId);
+        successEmails.push(emailArray[index]);
+      }
+    }
+    return { userIDArray, failedEmails, successEmails };
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+export const notificationsService = async (
+  data: notificationsRequest,
+  res: Response
+) => {
+  try {
+    const client: Client | undefined = await getDiscordBot();
+    const ids: getDiscordIDSchema | undefined = await getDiscordID(data.emails);
+    if (ids && client) {
```

Comment:
you should check for userIDArray to be empty. Not ids. `if(ids.userIDArray.length && client)` should work.